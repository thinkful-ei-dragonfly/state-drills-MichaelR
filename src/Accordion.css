.Accordion {
  margin: 20px;
}