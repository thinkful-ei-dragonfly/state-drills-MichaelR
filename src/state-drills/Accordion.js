import React from 'react';
import './Accordion.css';

class Accordian extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      openSection: null
    }
  }

  renderSection(section, i) {
    let content = '';
    if (i === this.state.openSection) {
      content = <p>{section.content}</p>
    }
    return (
      <li key={i}>
        <button onClick={this.handleClick} id={i}>{section.title}</button>
        {content}
      </li>
    )
  }

  handleClick = (e) => {
    let newIndex = parseInt(e.target.id)
    if (this.state.openSection === newIndex) {
      this.setState({
        openSection : null
      })
    } else {
      this.setState({
        openSection : newIndex
      })
    }
  }

  render() {
    let listItems
    if (this.props.sections) {
      listItems = this.props.sections.map((section, i) => this.renderSection(section, i))
    }
    return (
      <ul className="Accordion">
        {listItems}
      </ul>
    )
  }
}

// const sections = [
//   {
//     title: 'Section 1',
//     content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
//   },
//   {
//     title: 'Section 2',
//     content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
//   },
//   {
//     title: 'Section 3',
//     content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
//   },
// ]

// Accordian.defaultProps = ({sections});

export default Accordian;