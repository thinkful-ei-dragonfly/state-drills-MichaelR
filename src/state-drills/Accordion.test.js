import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-adapter-react-16';

import Accordion from './Accordion';

configure({adapter: new Adapter()});

const sections = [
  {
    title: 'Section 1',
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
  {
    title: 'Section 2',
    content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
  },
  {
    title: 'Section 3',
    content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
  },
]

describe('<Accordion />', () => {

  it('renders without crashing [snapshot test]', () => {
    // Render this component as JSON
    const tree = renderer.create(<Accordion />).toJSON();
    // Checks wether this matches the previous snapshot
    // The previous snapshot will be stored in __snapshots__/Accordion .test.js
    expect(tree).toMatchSnapshot();
  })

  it('renders empty given no sections', () => {
    // const wrapper = shallow(<Accordion />);
    // toJson(wrapper);
    // expect(toJson(wrapper)).toMatchSnapshot()
    const wrapper = shallow(<Accordion />);
    // toJson(wrapper);
    // expect(toJson(wrapper)).toMatchSnapshot()
  })
})