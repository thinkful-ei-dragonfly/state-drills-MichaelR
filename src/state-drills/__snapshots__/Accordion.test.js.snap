// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> renders without crashing [snapshot test] 1`] = `
<ul
  className="Accordion"
/>
`;

exports[`<Accordions /> renders without crashing [snapshot test] 1`] = `
<ul
  className="Accordion"
/>
`;
